tracer/src/Datadog.Trace/Activity/OtlpHelpers.cs

```diff
@@ -507,5 +616,52 @@ internal static void SerializeEventsToJson(Span span, IEnumerable events)
 
             return null;
         }
+
+        private static bool IsAllowedAtributeType(object? value)
+        {
+            if (value is null)
+            {
+                return false;
+            }
+
+            if (value is Array array)
+            {
+                if (array.Length == 0 ||
+                    array.Rank > 1)
+                {
+                    // Newtonsoft doesn't seem to support multidimensional arrays (e.g., [,]), but does support jagged (e.g., [][])
+                    return false;
+                }
+
+                if (value.GetType() is { } type
+                 && type.IsArray
+                 && type.GetElementType() == typeof(object))
+                {
+                    // Arrays may only have a primitive type, not 'object'
+                    return false;
+                }
+
+                value = array.GetValue(0)!;
```

Comment:
I don't think you need the dammit here right? 🤔 
```suggestion
                value = array.GetValue(0);
```